django/db/models/expressions.py

```diff
@@ -958,10 +958,13 @@ def resolve_all(child):
 
         def resolve(child):
             if hasattr(child, 'resolve_expression'):
-                return child.resolve_expression(
+                resolved = child.resolve_expression(
                     query=query, allow_joins=allow_joins, reuse=reuse,
                     summarize=summarize, for_save=for_save,
                 )
+                if hasattr(resolved, 'alias'):
+                    clone.queryset.query.external_aliases.add(resolved.alias)
```

Comment:
Maybe you want to add a code comment about that... something like, `# Add table aliases so they aren't unknown and thus quoted in the SQL.`